tensorrt_llm/_torch/pyexecutor/py_executor.py

```diff
@@ -1650,6 +1657,64 @@ def _update_requests(self, decoder_state: DecoderState):
             logger.error(f"Encountered an error in decode: {error_msg}")
             self._handle_errors(error_msg)
 
+    @nvtx_range("_generate_ngram_draft_tokens")
+    def _generate_ngram_draft_tokens(
+        self, scheduled_requests: ScheduledRequests
+    ) -> Tuple[ScheduledRequests, Dict[int, LlmRequest]]:
+        """
+        Prepares a batch for the draft model engine. Draft tokens are only produced
+        for generation requests.
+
+        The requests are prepared as follows:
+        1. The first time the draft engine sees a request, it's a context request.
+        2. Otherwise, if draft tokens were accepted on the last target model decoding
+        step, it's a chunked context request (we process all the accepted tokens together).
+        3. Otherwise, it's a generation request.
```

Comment:
Duplicated comment? Seems to not apply to this function

Reply:
Updated comment